lib/index.ts

```diff
@@ -16,23 +16,33 @@
  */
 
 import * as _ from 'lodash';
+import type * as Stream from 'stream';
 
-import type { DockerVersion, ImagePushOptions } from 'dockerode';
+import type {
+	DockerVersion,
+	ImagePushOptions,
+	ImageBuildOptions,
+	AuthConfig,
+} from 'dockerode';
 
 export interface EngineVersion extends DockerVersion {
 	Engine?: string;
 }
 
 export interface PullPushOptions {
-	authconfig?: {
-		username?: string;
-		password?: string;
+	ignoreProgressErrorEvents?: boolean;
+	authconfig?: AuthConfig & {
 		auth?: string;
-		email?: string;
-		serveraddress?: string;
 		registrytoken?: string;
```

Comment:
`auth` and `registrytoken` exist in the Auth object in the Docker engine https://github.com/moby/moby/blob/master/api/types/auth.go but they are not documented as part of the Docker API https://docs.docker.com/engine/api/v1.37/#operation/SystemAuth 

The supervisor uses `registrytoken` to authenticate to the registry when pulling images with deltas